src/Interpreters/executeQuery.cpp

```diff
@@ -316,9 +316,14 @@ QueryLogElement logQueryStart(
 
     bool log_queries = settings.log_queries && !internal;
 
+    if (!context->getQueryLog())
+        return elem;
+
     /// Log into system table start of query execution, if need.
     if (log_queries)
     {
+        auto query_log = context->getQueryLog();
```

Comment:
Guess you could call `context->getQueryLog()` just once (instead of in l. 319/325)